### PR TITLE
Add named expressions to UCR data sources

### PR DESCRIPTION
**Pull request by czue**
This was inspired by @dannyroberts's implementation of named filters, and getting overwhelmed by the amount of repetition in some of the ICDS data sources and thinking this would be a very simple and useful thing to add.

@NoahCarnahan @nickpell @sravfeyn @esoergel and @sheelio might be interested. cc @proteusvacuum

### CODE REVIEW

File: corehq/apps/userreports/README.md [hunk 1]
```diff
@@ -303,6 +304,34 @@ Here is a simple example that demonstrates the structure. The keys of `propertie
 }
 ```
 
+#### Named Expressions
+
+Last, but certainly not least, are named expressions.
+These are special expressions that can be defined once in a data source and then used throughout other filters and indicators in that data source.
+This allows you to write out a very complicated expression a single time, but restill use it in multiple places with a simple syntax.
```
Comment (snopoke) [inline corehq/apps/userreports/README.md]: spelling `restill`

Comment (czue) [inline corehq/apps/userreports/README.md]: :eyes: fixed

**Comment by sheelio** [comment]
@czue This is awesome - super useful.

File: corehq/apps/userreports/tests/test_data_source_config.py [hunk 2]
```diff
@@ -129,6 +130,143 @@ def test_doc_type_is_required(self):
...
+                    'test': {
+                        'type': 'boolean_expression',
+                        'expression': {
+                            'type': 'property_name',
+                            'property_name': 'is_evil',
+                        },
+                        'operator': 'eq',
+                        'property_value': True,
+                    },
+                    'expression_if_true': "mwa-ha-ha",
```
Comment (snopoke) [inline corehq/apps/userreports/tests/test_data_source_config.py]: lol

Comment (snopoke) [inline corehq/apps/userreports/tests/test_data_source_config.py]: or should I say `mwa-ha-ha` :smiling_imp:

**Comment by snopoke** [comment]
nice

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**